pkg/csi/service/vanilla/controller.go

```diff
@@ -153,10 +158,25 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 	err = common.CheckAPI(vc.Client.ServiceContent.About.ApiVersion, common.MinSupportedVCenterMajor,
 		common.MinSupportedVCenterMinor, common.MinSupportedVCenterPatch)
 	if err != nil {
-		log.Errorf("checkAPI failed for vcenter API version: %s, err=%v", vc.Client.ServiceContent.About.ApiVersion, err)
+		log.Errorf("checkAPI failed for vcenter API version: %s, err=%v",
+			vc.Client.ServiceContent.About.ApiVersion, err)
 		return err
 	}
 
+	csiSnapshotFSSEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.BlockVolumeSnapshot)
+	if csiSnapshotFSSEnabled {
+		// Check VC support only if internal FSS is enabled.
+		err = common.CheckAPI(vc.Client.ServiceContent.About.ApiVersion, common.SnapshotSupportedVCenterMajor,
+			common.SnapshotSupportedVCenterMinor, common.SnapshotSupportedVCenterPatch)
+		if err != nil {
+			log.Warnf("checkAPI failed for snapshot support on vCenter API version: %s, will not be "+
+				"reporting snapshot capabilities, err=%v", vc.Client.ServiceContent.About.ApiVersion, err)
```

Comment:
Is this consistent with other failed capability check  due to the vCenter connectivity failure?

Reply:
Yes, by the time we call `CheckAPI`, we already got the the connected vc info, and we simply pass the version `vc.Client.ServiceContent.About.ApiVersion` to `CheckAPI` method.

If there was an issue getting a connected vc, the csi driver wouldn't even startup as `Init` would return an err: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/vanilla/controller.go#L121-L125

Reply:
We should error it out here. Just like the check.

```
isvSANFileServicesSupported, err := c.manager.VcenterManager.IsvSANFileServicesSupported(ctx,
			c.manager.VcenterConfig.Host)
if err != nil {
	log.Errorf("failed to verify if vSAN file services is supported or not. Error:%+v", err)
	return err
}
```